src/NetworkPaymentAdapter.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity 0.8.13;
+
+import {INetworkTreasury} from "./INetworkTreasury.sol";
+
+interface VestLike {
+    function vest(uint256) external;
+    function unpaid(uint256) external view returns (uint256);
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (address);
+    function join(address, uint256) external;
+}
+
+interface DaiLike {
+    function balanceOf(address) external view returns (uint256);
+    function transfer(address, uint256) external returns (bool);
+}
+
+/// @title Payment adapter to the keeper network treasury
+/// @dev Sits between dss-vest and the keeper network treasury contract
+contract NetworkPaymentAdapter {
```

Comment:
+ IFace ?

Reply:
I can add that if it helps. This isn't usual practise for Maker though.